freud/locality.pxd

```diff
@@ -28,14 +65,15 @@ cdef class NeighborQueryResult:
     # compile with a cdef method.
     @staticmethod
     cdef inline NeighborQueryResult init(
-            freud._locality.NeighborQuery * nqptr, const float[:, ::1] points,
+            NeighborQuery nq, const float[:, ::1] points,
             cbool exclude_ii, float r=0, unsigned int k=0):
         # Internal API only
         assert r != 0 or k != 0
 
         obj = NeighborQueryResult()
 
-        obj.nqptr = nqptr
+        obj.nq = nq
+        obj.nqptr = nq.nqptr
```

Comment:
I would prefer not to store both of these, if we're going to store the Cython object let's just access the `nqptr` through `self.nq`.

Reply:
This also applies to the `AABBQueryResult` (removing the aabbptr class attribute).